src/workerd/util/sqlite.c++

```diff
@@ -445,6 +445,43 @@ kj::Own<sqlite3_stmt> SqliteDatabase::prepareSql(
   }
 }
 
+kj::StringPtr SqliteDatabase::ingestSql(Regulator& regulator, kj::StringPtr sqlCode) {
+  // While there's still some input SQL to process
+  while (sqlCode.begin() != sqlCode.end()) {
+    // And there are still valid statements:
+    auto statementLength = sqlite3_complete_length(sqlCode.begin(), 1);
+    if (!statementLength) break;
+
+    // Slice off the next valid statement and prepare it
+    auto nextStatement = kj::str(sqlCode.slice(0, statementLength));
+    auto result = prepareSql(regulator, nextStatement, 0, SINGLE);
```

Comment:
Instead of calling `prepareSql()` directly, I think you should construct a `Query` here and then call it.

* That will take care of the write callback for you.
* It'll also take care of a few things you are missing here, like setting `currentRegulator` correctly. It is actually necessary to set the regulator when _executing_ the query, in addition to preparing it, since various things can cause the query to be rebuilt at runtime.

Reply:
This is now done in commit https://github.com/cloudflare/workerd/commit/624db515d1ea53d947cb0e7f2b8a4c4f9916e383

Interestingly creating the `Query` is sufficient since it calls `init` which calls `nextRow` internally, so I don't actually have to do anything with the query object for the statement to be executed. Seems a bit weird to write:

```cpp
// Create a Query object, which will prepare & execute it
Query(*this, regulator, nextStatement);
```

without assigning it to a variable. Is there a different way I should express that (like `auto _query = Query(...)`?)

Reply:
Casting to void is the C++ idiom for "I am intentionally ignoring this result", so something like

```
// Create a Query object, which will prepare & execute it
(void)Query(*this, regulator, nextStatement);
```

This only matters to the compiler if `Query` is marked `[[nodiscard]]`, which it isn't, but it will communicate to human readers of the code that yes, we're making and discarding this Query, and no, we didn't forget to do something with it.